src/charm.py

```diff
@@ -395,12 +395,18 @@ def _configure_pebble(self, restart: bool = False) -> None:
         Args:
             restart (bool): Whether to restart the Pebble service. Defaults to False.
         """
-        self._container.add_layer(self._container_name, self._pebble_layer, combine=True)
+        plan = self._container.get_plan()
+        if plan.services != self._pebble_layer.services:
+            self._container.add_layer(
+                self._container_name, self._pebble_layer, combine=True
+            )
+            self._container.replan()
+            self.logger.info("New layer added: %s", self._pebble_layer)
+            restart = True
```

Comment:
The replan will restart what it needs to, so we do not need to force a restart here, simply let it fall through to the conditional below.